### PR TITLE
Add feature flag to just completely skip sync and symlink operations

### PR DESCRIPTION
**Pull request by agjohnson**
We can control this on a per-project basis. Once we remove symlinking
and Syncers, this can go away entirely.

### CODE REVIEW

**Comment by agjohnson** [comment]
I'm not confident I protected all of the sync/symlink calls. I didn't put the same logic in the actual tasks as we shouldn't fire off thousands of tasks that are effectively noop tasks. This did save the webs from being hammered with thousands of symlink tasks for the project however.

**Comment by davidfischer** [comment]
This worked for me in my testing to stop symlinking versions for specific projects. This didn't protect the `symlink_subproject` function but those subprojects would technically be different projects and could be feature flagged themselves. :+1:

File: readthedocs/projects/models.py [hunk 1]
```diff
@@ -1505,6 +1505,7 @@ def add_features(sender, **kwargs):
     UPDATE_CONDA_STARTUP = 'update_conda_startup'
     CONDA_APPEND_CORE_REQUIREMENTS = 'conda_append_core_requirements'
     ALL_VERSIONS_IN_HTML_CONTEXT = 'all_versions_in_html_context'
+    SKIP_SYNC = 'skip_sync'
```
Comment (humitos) [inline readthedocs/projects/models.py]: nit: I would call it `SKYP_WEB_SERVERS_SYNC` to make it explicit.

**Comment by humitos** [comment]
I don't think it worth the effort, but we could enable this Feature flag automatically on `Project.save` when we detect that `skip = True` and remove it otherwise.

**Pull request merged**